src/compiler/expression/query.rs

```diff
@@ -20,22 +20,27 @@ impl Query {
     // TODO:
     // - error when trying to index into object
     // - error when trying to path into array
+    #[must_use]
```

Comment:
Perhaps it's better to apply `#[must_use]` to the `struct Query`.

Reply:
Does that mean the same thing? These were from `clippy` suggestions (I'm not sure why it wanted it on these functions all of a sudden)

Reply:
I tried it, unfortunately Clippy starts complaining if I only put it on the struct itself :disappointed:

Reply:
Ah I see, thanks for trying it out. This might happen because [std::hint::must_use](https://doc.rust-lang.org/std/hint/fn.must_use.html) is still experimental.